staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -88,7 +88,9 @@ func StreamObject(statusCode int, gv schema.GroupVersion, s runtime.NegotiatedSe
 // The context is optional and can be nil. This method will perform optional content compression if requested by
 // a client and the feature gate for APIResponseCompression is enabled.
 func SerializeObject(mediaType string, encoder runtime.Encoder, hw http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
+	auditId, _ := request.AuditIDFrom(req.Context())
```

Comment:
I'm not a fan of ignoring the second argument (i.e. whether auditID is set).
Shouldnt' we set the value to "<unknown>" or sth like that If AuditIDFrom returns false as a second param?

Reply:
Fair enough.

While refactoring, I found some inconsistency - there was already a function: [GetAuditIDTruncated(req *http.Request)](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/request/auditid.go#L52) which would not only do the check on the second argument and return "" if it's not found (which is exactly what we wanted), but it's truncating an audit-id string, if it's above 64 chars. And this function is already used in other logs/traces.

I believe it's important to be consistent across logs (not to have full audit-id in one place and truncated in another), therefor, I've refactor the base function a bit, so it's based on `context` rather than `request`, so it could be reused in new files: catcher and store.go.